python/dgl/distributed/kvstore.py

```diff
@@ -1283,6 +1343,38 @@ def _take_id(self, elem):
         """
         return elem.server_id
 
+    def count_nonzero(self, name):
+        """Count nonzero value by pull request from KVServers.
+
+        Parameters
+        ----------
+        name : str
+            data name
+
+        Returns
+        -------
+        int
+            the number of nonzero in this data.
+        """
+        total = 0
+        pull_count = 0
+        for machine_id in range(self._machine_count):
+            if machine_id == self._machine_id:
+                local_id = F.tensor(np.arange(self._part_policy[name].get_part_size(),
+                                              dtype=np.int64))
+                local_data = self._pull_handlers[name](self._data_store, name, local_id)
```

Comment:
do you need to have a data copy here? since this is local machine, you can access the local tensor directly.

Reply:
Yes, I could have used the simpler method you said. 
But in the ”pull data function“, it uses local_data in such way. I guess may be there some other reasons for this?
If possible, I would prefer to use local_data directly.

Reply:
thanks